File: src/main/java/de/ayont/lpc/AutoAnnouncer.java
package de.ayont.lpc;

import net.kyori.adventure.text.minimessage.MiniMessage;
import org.bukkit.scheduler.BukkitRunnable;

import java.util.List;
import java.util.Random;
import java.util.concurrent.atomic.AtomicInteger;

public class AutoAnnouncer extends BukkitRunnable {

    private final LPC plugin;
    private final MiniMessage miniMessage;
    private final AtomicInteger currentIndex = new AtomicInteger(0);
    private final Random random = new Random();

    public AutoAnnouncer(LPC plugin) {
        this.plugin = plugin;
        this.miniMessage = MiniMessage.miniMessage();
    }

    @Override
    public void run() {
        if (!plugin.getConfig().getBoolean("announcer.enabled", false)) {
            return;
        }

        List<String> messages = plugin.getConfig().getStringList("announcer.messages");
        if (messages.isEmpty()) {
            return;
        }

        String strategy = plugin.getConfig().getString("announcer.selection-strategy", "ordered");
        if (strategy == null) strategy = "ordered";
        strategy = strategy.toLowerCase();

        String messageToSend;
        int size = messages.size();

        switch (strategy) {
            case "random":
                messageToSend = messages.get(random.nextInt(size));
                break;
            case "reverse":
                 int revIdx = currentIndex.decrementAndGet();
                 if (revIdx < 0 || revIdx >= size) {
                     revIdx = size - 1;
                     currentIndex.set(revIdx);
                 }
                 messageToSend = messages.get(revIdx);
                 break;
            case "ordered":
            default:
                int ordIdx = currentIndex.getAndIncrement();
                if (ordIdx < 0 || ordIdx >= size) {
                    ordIdx = 0;
                    currentIndex.set(1);
                }
                messageToSend = messages.get(ordIdx);
                break;
        }

        // Broadcast to all players using Adventure
        plugin.getAdventure().all().sendMessage(miniMessage.deserialize(messageToSend));
    }
}

File: src/main/kotlin/de/ayont/lpc/AutoAnnouncer.kt
package de.ayont.lpc

import net.kyori.adventure.text.minimessage.MiniMessage
import org.bukkit.scheduler.BukkitRunnable
import java.util.*
import java.util.concurrent.atomic.AtomicInteger

class AutoAnnouncer(private val plugin: LPC) : BukkitRunnable() {

    private val miniMessage = MiniMessage.miniMessage()
    private val currentIndex = AtomicInteger(0)
    private val random = Random()

    override fun run() {
        if (!plugin.config.getBoolean("announcer.enabled", false)) {
            return
        }

        val messages = plugin.config.getStringList("announcer.messages")
        if (messages.isEmpty()) {
            return
        }

        val strategy = plugin.config.getString("announcer.selection-strategy", "ordered")?.lowercase() ?: "ordered"

        val size = messages.size
        val messageToSend = when (strategy) {
            "random" -> messages[random.nextInt(size)]
            "reverse" -> {
                var revIdx = currentIndex.decrementAndGet()
                if (revIdx < 0 || revIdx >= size) {
                    revIdx = size - 1
                    currentIndex.set(revIdx)
                }
                messages[revIdx]
            }
            else -> {
                var ordIdx = currentIndex.getAndIncrement()
                if (ordIdx < 0 || ordIdx >= size) {
                    ordIdx = 0
                    currentIndex.set(1)
                }
                messages[ordIdx]
            }
        }

        // Broadcast to all players using Adventure
        plugin.adventure?.all()?.sendMessage(miniMessage.deserialize(messageToSend))
    }
}
